sympy/integrals/transforms.py

```diff
@@ -1008,9 +1008,34 @@ def repl(ex, *args):
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, limit,
+                       DiracDelta)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    zero_minus = symbols('zero_minus', real=True)
+    # Expressions with DiracDelta need to be computed with a directional
+    # limit, but this will fail on some sums, so we first use the
+    # linearity of the Laplace transform to split the integral into a sum
+    # of integrals.
+    if f.has(DiracDelta):
+        f = f.expand(deep=True, modulus=None, power_base=False,
+                     power_exp=False, mul=True, log=False, multinomial=True,
+                     basic=True, complex=False, func=False, trig=False)
+        if f.is_Add:
+            _all_parts = []
+            for _part in f.args:
+                if _part.has(DiracDelta(t)):
+                    _all_parts.append(limit(integrate(exp(-s*t) * _part,
```

Comment:
It is inefficient to invoke `limit -> integrate -> Integral -> doit -> ...` separately for each term, in particular, if `expand` creates a great number of terms. It would be better to collect all terms is two lists, one with `DiracDelta`s and the other with the rest, and then integrate the sums (`Add`) of the lists. Those lists can be created with `sift`.

Reply:
I am aware of the inefficiency, but did it on purpose. I tried your idea first, exactly for efficiency reasons, but found too many cases where `limit`'s `mrv` would not work, but fail with an error. So I chose inefficiency over crashing.

I think this is an OK trade-off, because one of those maybe many terms contains only one Delta function and is not costly to evaluate.

I see in the issues that `mrv` causes other problems too, so maybe we should revisit this discussion when `mrv` is fixed.

Would that be OK for you?

Reply:
Do we need limit at all?

My suggestion for lineq2dict above decomposes the integrand into a linear combination of DiracDelta with coefficients that are not DiracDelta so we have
```python
a(t) + f_1(t)*δ(g_1(t)) + f_2(t)*δ(g_2(t)) + ...
```
The reason for doing this was just to isolate δ terms that have `g_i(0) = 0` because those are the cases not currently handled correctly by integrate (without using limit). Then we can compute the integral as before but we can add the extra results from some of the δ terms.

All those DiracDelta terms with `g_i(0) = 0` give additional terms that we don't get automatically from integrate. We need to add those terms on and that means adding e.g. `f_i(0) / |g_i'(0)|` provided `g_i'(0) != 0`.

Reply:
> too many cases where limit's mrv would not work,

Can you give examples? Those should be investigated.

Reply:
> > too many cases where limit's mrv would not work,
> 
> Can you give examples? Those should be investigated.

Sorry, I did not do that professionally enough. I cannot reconstruct those cases immediately. However, if I ever find anything again, I will submit an issue.

Reply:
> Do we need limit at all?

I think not.

> My suggestion for lineq2dict above decomposes the integrand into a linear combination of DiracDelta with coefficients [...]
> All those DiracDelta terms with `g_i(0) = 0` give additional terms that we don't get automatically from integrate. We need to add those terms on and that means adding e.g. `f_i(0) / |g_i'(0)|` provided `g_i'(0) != 0`.

Brilliant idea! It was not very difficult to integrate, we just have to catch derivatives of `DiracDelta` if we do this. I'll push once my own tests have run through.